### ISSUE TITLE
Allowed to log in  as a deleted user to a github auth enabled setup.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#2021]
Server version -  V0.37.0-rc1.

github auth enabled setup.
Logged to server as user - "user1" which had access to "Default" project.

As admin user,
I used the "Deactivate" option to deactivate "user1" account which gets to "inactive" state and the deleted this account.

I am still being allowed to log into rancher-server as "user1" account.

Expected Behavior:
If an account is deleted , i should not be allowed to access server anymore as this account.

Also these deleted account continued to be listed as members of environment it was part of before deletion. 

```
 select * from account;
| 52 | Sidhartha Mani        | user       | b3ed60d9-1642-4950-975e-e22488284ec9 | NULL        | purged   | 2015-09-10 21:29:16 | 2015-09-10 23:15:39 | 2015-09-10 23:16:39 | {"fields":{"hasLoggedIn":true,"githubaccess_token":"7cb15fc113cec86e8624aa0dfe6c3beba01415df"}} | 1050521     | github_user      |

select * from project_member;
|  9 | Sidhartha Mani      |          5 | projectMember | 554c816e-84eb-429d-bbe7-d646689de9a6 | NULL        | active | 2015-09-10 21:28:26 | NULL                | NULL                | {}   | 1050521     |          5 | github_user      | member |
+----+---------------------+------------+---------------+--------------------------------------+-------------+--------+---------------------+---------------------+---------------------+------+-------------+------------+------------------+--------+

```

**Comment by hibooboo2** [comment]
Addressed with rancher/cattle#860

**Comment by sangeethah** [comment]
Tested with build from master - Oct 8

In a  github auth enabled setup.
Logged to server as user - "user1" for  which  access was provided to few of existing environments.

As admin user,
Used the "Deactivate" option to deactivate "user1" account.

For a while (until the user is purged) , we are not allowed to log in to rancher server as user1.

Once the user is purged , we are  allowed to log in to rancher server as user1.
There is a new account created at this point which has no access to any existing environments.

When user logs in , a new default environment gets created for this user which is the only environment he has access to.
